### PR TITLE
DONTMERGE: This is just a test PR for sorald-buildbreaker

### PR DESCRIPTION
**Pull request by slarse**
I'm just trying to implement PR commenting, see spoonlabs/sorald-buildbreaker#53

### CODE REVIEW

File: src/main/java/sorald/Main.java [hunk 1]
```diff
@@ -4,6 +4,12 @@
 
 public class Main {
     public static void main(String[] args) {
+        try {
+            throw new InterruptedException();
+        } catch (InterruptedException | IllegalArgumentException e) {
+            System.out.println("oops!");
+        }
+
```
Comment (slarse) [inline src/main/java/sorald/Main.java:12]:
```suggestion
this is better
```

Comment (slarse) [reply src/main/java/sorald/Main.java:12]: HAH, it worked :D

Comment (slarse) [inline src/main/java/sorald/Main.java:12]:
```suggestion
this is better
```

Comment (monperrus) [reply src/main/java/sorald/Main.java:12]: oh super nice!

Comment (slarse) [inline src/main/java/sorald/Main.java:12]:
```suggestion
this is better
```

File: src/main/java/sorald/Main.java [hunk 2]
```diff
@@ -4,6 +4,12 @@
 
 public class Main {
     public static void main(String[] args) {
+        try {
+            throw new InterruptedException();
+        } catch (InterruptedException | IllegalArgumentException e) {
+            System.out.println("oops!");
```
Comment (slarse) [inline src/main/java/sorald/Main.java:10]: To fix violation '2142:sorald/Main.java:9:17:9:66', Sorald suggests the following:
```suggestion
$(hunk.additions.join('
'))
```

[the remark this answers is not in the document]
Comment (slarse) [reply src/main/java/sorald/Main.java:10]: I mean, that's not right, but we're getting somewhere :D

**Comment by slarse** [comment]
For some flabbergasting reason, `git diff` produced the following diff on GHA:

```diff
 /usr/bin/git diff -U0
diff --git a/src/main/java/sorald/Main.java b/src/main/java/sorald/Main.java
index 0af4c5d..108992f 100644
--- a/src/main/java/sorald/Main.java
+++ b/src/main/java/sorald/Main.java
@@ -9 +9 @@ public class Main {
-        } catch (InterruptedException | IllegalArgumentException e) {
+        } catch (java.lang.InterruptedException | java.lang.IllegalArgumentException e) {
@@ -10,0 +11,3 @@ public class Main {
+            if (e instanceof InterruptedException)
+                Thread.currentThread().interrupt();
+
```

I have no idea of why it makes two hunks out of that, it doesn't happen for me locally.

**Comment by slarse** [comment]
@monperrus AHA, great success! See the two suggestions posted above by the GHA bot. It's not a bug in buildbreaker that one fix got posted as two suggestions, because the patch actually looks like this with sorald-0.1.0 (which is still the default in buildbreaker):

```diff
             throw new InterruptedException();
-        } catch (InterruptedException | IllegalArgumentException e) {
+        } catch (java.lang.InterruptedException | java.lang.IllegalArgumentException e) {
             System.out.println("oops!");
+            if (e instanceof InterruptedException)
+                Thread.currentThread().interrupt();
+
         }
 
         int exitStatus = Cli.createCli().execute(args);
```

While sorald-0.2.0 performs this patch instead:

```diff
             throw new InterruptedException();
         } catch (InterruptedException | IllegalArgumentException e) {
             System.out.println("oops!");
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
         }
 
         int exitStatus = Cli.createCli().execute(args);
```

While we should of course use the latest version of Sorald in buildbreaker, this does highlight one thing we will want to do: merge diff hunks that are very close and relate to the same violation specifier. I.e. we should strive for one suggestion per violation specifier, even if that violation spawns multiple diff hunks.

**Comment by algomaster99** [comment]
Closing this due to inactivity and I think its purpose has been served.

**Pull request closed without merging**